### PR TITLE
fix(core): edge-case memory leak during disk failure fixed

### LINKED ISSUE #5348: failing fuzz test - WalWriterFuzzTest.testAddDropColumnDropPartition()
### To reproduce

Run `WalWriterFuzzTest.testAddDropColumnDropPartition()` with these seeds: 
```
        s0 = 872206675813L;
        s1 = 1738242160678L;
```

QuestDB commit: 42ebd0a69f923c4
The test is failing consistently with:
```
java.lang.AssertionError: file descriptors, expected: [4294967296,103079215107,98784247813,107374182408], actual: [4294967296,103079215107,98784247813,472446404878,476741372175,481036339472,523986012433,107374193863] 
Expected :3
Actual   :7
<Click to see difference>
	at junit@4.13.2/org.junit.Assert.fail(Assert.java:89)
	at junit@4.13.2/org.junit.Assert.failNotEquals(Assert.java:835)
	at junit@4.13.2/org.junit.Assert.assertEquals(Assert.java:647)
	at io.questdb.test/io.questdb.test.tools.TestUtils$LeakCheck.close(TestUtils.java:1999)
	at io.questdb.test/io.questdb.test.tools.TestUtils.assertMemoryLeak(TestUtils.java:670)
	at io.questdb.test/io.questdb.test.AbstractCairoTest.assertMemoryLeak(AbstractCairoTest.java:1061)
	at io.questdb.test/io.questdb.test.cairo.fuzz.AbstractFuzzTest.runFuzz(AbstractFuzzTest.java:191)
	at io.questdb.test/io.questdb.test.cairo.fuzz.WalWriterFuzzTest.runFuzz(WalWriterFuzzTest.java:557)
	at io.questdb.test/io.questdb.test.cairo.fuzz.WalWriterFuzzTest.testAddDropColumnDropPartition(WalWriterFuzzTest.java:100)
[...]
```

### QuestDB version:

8.2.2

### OS, in case of Docker specify Docker and the Host OS:

Linux@Intel

### File System, in case of Docker specify Host File System:

ext4

### Full Name:

Jaromir Hamala

### Affiliation:

QuestDB

### Have you followed Linux, MacOs kernel configuration steps to increase Maximum open files and Maximum virtual memory areas limit?

- [x] Yes, I have

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jerrinot**
fixes #5348

the issue found by a fuzz test

### CODE REVIEW

**Comment by glasstiger** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|2|2|100.00%|

File: core/src/main/java/io/questdb/cairo/TableWriter.java [hunk 1]
```diff
@@ -4234,11 +4234,12 @@ private void createSymbolMapWriter(CharSequence name, long columnNameTxn, int sy
                 denseSymbolMapWriters.size(),
                 txWriter
         );
+        denseSymbolMapWriters.add(w);
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/cairo/TableWriter.java:4237]: i wonder if we should just close the writer there and then and let exception fly? Having distressed writer in the data structures is no good

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

File: core/src/main/java/io/questdb/cairo/TableWriter.java (hunk 1 above)
Comment (jerrinot) [reply core/src/main/java/io/questdb/cairo/TableWriter.java:4249]: @bluestreak01 wdyt now?

**Comment by glasstiger** [comment]
#### [PR Coverage check]

:disappointed: **fail** : 2 / 5 (40.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|2|5|40.00%|

**Review by bluestreak01 — APPROVED** [review]
